test/18-startup.ts

```diff
@@ -1,10 +1,34 @@
+import { SinonStub, stub } from 'sinon';
+
+import APIBinder from '../src/api-binder';
+import * as constants from '../src/lib/constants';
 import { Supervisor } from '../src/supervisor';
 import { expect } from './lib/chai-config';
 
 describe('Startup', () => {
-	it('should startup correctly', function() {
+	let initClientStub: SinonStub;
+	let startStub: SinonStub;
+	let vpnStatusPathStub: SinonStub;
+
+	before(() => {
+		initClientStub = stub(APIBinder.prototype as any, 'initClient').returns(
+			Promise.resolve(),
+		);
+		startStub = stub(APIBinder.prototype as any, 'start').returns(
+			Promise.resolve(),
+		);
+		vpnStatusPathStub = stub(constants, 'vpnStatusPath').returns('');
+	});
+
+	after(() => {
+		initClientStub.restore();
+		startStub.restore();
+		vpnStatusPathStub.restore();
+	});
+
+	it('should startup correctly', async () => {
 		const supervisor = new Supervisor();
-		expect(supervisor.init()).to.not.throw;
+		expect(await supervisor.init()).to.not.throw;
```

Comment:
```suggestion
		expect(async () => await supervisor.init()).to.not.throw;
```

Reply:
When the expect is written like that the exception thrown inside that closure doesn't trigger the test to fail. If this comment was related to the failing tests the issue was my local machine had docker socket open so the test passed but in CI docker is not there so it fails. Just another thing to stub.